src/components/EntityInfo/EntityMenu.vue

```diff
@@ -0,0 +1,186 @@
+<template>
+  <menu id="entity-menu">
+    <!-- Staff Comments -->
+    <span v-if="!!businessId && isAllowed(AllowableActions.ADD_STAFF_COMMENT)">
+      <StaffComments
+        :axios="axios"
+        :businessId="businessId"
+        :maxLength="2000"
+      />
+    </span>
+
+    <span v-if="isHistorical">
+      <v-chip
+        id="historical-chip"
+        class="primary mt-n1 ml-4 pointer-events-none" small label text-color="white"
+      >
+        <strong>HISTORICAL</strong>
+      </v-chip>
+      <span class="font-14 mx-3">{{ getReasonText || 'Unknown Reason' }}</span>
+    </span>
+
+    <template v-else>
+      <!-- View and Change Company Information -->
+      <span v-if="isAllowed(AllowableActions.VIEW_CHANGE_COMPANY_INFO)">
+        <v-btn
+          small text color="primary"
+          id="company-information-button"
+          :disabled="hasBlocker"
+          @click="promptChangeCompanyInfo()"
+        >
+          <v-icon medium>mdi-file-document-edit-outline</v-icon>
+          <span class="font-13 ml-1">View and Change Business Information</span>
+        </v-btn>
+
+        <v-tooltip v-if="isPendingDissolution" top content-class="top-tooltip" transition="fade-transition">
+          <template v-slot:activator="{ on }">
+            <v-icon color="orange darken-2" size="24px" class="pr-2" v-on="on">mdi-alert</v-icon>
+          </template>
+          You cannot view or change business information while the business is pending dissolution.
+        </v-tooltip>
+      </span>
+
+      <!-- Dissolve Company -->
+      <span v-if="isAllowed(AllowableActions.DISSOLVE_COMPANY)">
+        <v-tooltip top content-class="top-tooltip" transition="fade-transition">
+          <template v-slot:activator="{ on }">
+            <v-btn
+              small text color="primary"
+              id="dissolution-button"
+              :disabled="hasBlocker"
+              @click="promptDissolve()"
+              v-on="on"
+            >
```

Comment:
This button and several other buttons in this component appear to also have a long chains of emits that are passed back to the parent and then the grandparent.  Can we either reduce the length of the chains or create tech debt tickets to refactor this?